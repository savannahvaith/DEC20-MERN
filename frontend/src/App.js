import Navigation from './Components/Header/Navigation';
import './Resources/App.css';

import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import HomePage from './Components/Header/HomePage';
import About from './Components/Header/About';

function App() {
  return (
    <Router>
      <Navigation/>
      <Switch>
      <Route path="/" exact>
          <HomePage/>
      </Route>
      <Route path="/about">
          <About/>
      </Route>
      <Route>
        {/* Call a component for a 404 NOT FOUND */}
        <p>Sorry can't find what you're looking for!</p>
      </Route>
      </Switch>
    </Router>
  );
}

export default App;
