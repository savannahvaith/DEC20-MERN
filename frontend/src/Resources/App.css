body { 
  background-color: darkcyan!important;
  overflow-x: hidden;
}

#sidebar{
  min-height: 100vh; 
  margin-left: 0px;
}