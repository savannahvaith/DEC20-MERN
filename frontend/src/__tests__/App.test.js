import { render, screen } from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom';
import { expect, it } from '@jest/globals';
import {create} from 'react-test-renderer';
// snapshot test
// describe(`Snap shot test of APP`, () => {

  test('renders learn react link', () => {
    render(<App />);
    const linkElement = screen.getByText(/github/i);
    expect(linkElement).toBeInTheDocument();
  });


  it(`It should match the snapshot every time it's rendered`, () => {
    const instance = create(<App/>).toJSON(); // returns an object representing the rendered TREE!
    expect(instance).toMatchSnapshot(); 
    // i don't have an initial picture of the tree (component tree)
    // it will create a snapshot for the firsttime if it doesn't already exist
    // every subseuqnt time we we refer to the snapshot it will use that snapshot as reference!
  })
// });
