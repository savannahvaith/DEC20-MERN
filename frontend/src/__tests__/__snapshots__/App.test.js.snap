// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should match the snapshot every time it's rendered 1`] = `
<div
  className="container"
>
  <div>
    <nav
      className="navbar navbar-expand-md navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        DecSoftware2020
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="mr-auto navbar-nav"
        >
          <li
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              href="#"
              onClick={[Function]}
            >
              Components
            </a>
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                <a
                  className="nav-link"
                  href="/Prop"
                  onClick={[Function]}
                >
                  Props
                </a>
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                <a
                  className="nav-link"
                  href="/User"
                  onClick={[Function]}
                >
                  Static Data
                </a>
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                <a
                  className="nav-link"
                  href="/States"
                  onClick={[Function]}
                >
                  States
                </a>
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                <a
                  className="nav-link"
                  href="Lifting-States/Searchable"
                  onClick={[Function]}
                >
                  Lifting States (Searchable List)
                </a>
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                <a
                  className="nav-link"
                  href="/Lifting-States/ProductManager"
                  onClick={[Function]}
                >
                  Lifting States (Product Manager)
                </a>
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                <a
                  className="nav-link"
                  href="/UseEffect"
                  onClick={[Function]}
                >
                  UseEffect-Hooks
                </a>
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                <a
                  className="nav-link"
                  href="/OMDB/Search"
                  onClick={[Function]}
                >
                  OMDB
                </a>
              </button>
            </div>
          </li>
        </ul>
        <a
          className="nav-link"
          href="https://github.com/savannahvaith/DEC20-MERN"
          onClick={[Function]}
        >
          GitHub
        </a>
      </div>
    </nav>
  </div>
  <h1>
     Welcome to the website! I added this!!!!
  </h1>
</div>
`;
